### PR TITLE
feat: default to Apollo reporting over OTel

### PR DESCRIPTION
**Pull request by lennyburdette**
BREAKING CHANGE: This change will cause traces in Apollo Studio to appear differently

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: .changesets/breaking_lb_next_apollo_otel_default.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+### feat: default to Apollo reporting over OTel ([PR #5741](https://github.com/apollographql/router/pull/5741))
```
Comment (dylan-apollo) [inline .changesets/breaking_lb_next_apollo_otel_default.md:1]:
```suggestion
### Default to Apollo reporting over OTel ([PR #5741](https://github.com/apollographql/router/pull/5741))
```

File: .changesets/breaking_lb_next_apollo_otel_default.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+BREAKING CHANGE: This change will cause traces in Apollo Studio to appear differently
```
Comment (dylan-apollo) [inline .changesets/breaking_lb_next_apollo_otel_default.md:3]: Should we include the config snippet to turn this back off?

File: .changesets/breaking_lb_next_apollo_otel_default.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+<!-- https://apollographql.atlassian.net/browse/CNN-330 -->
```
Comment (dylan-apollo) [inline .changesets/breaking_lb_next_apollo_otel_default.md:5]: I assume we don't want this in the changelog—not positive, though

**Review by dylan-apollo — COMMENTED** [review]
Makes sense to me, pretty sure I heard agreement on this change in a previous meeting

File: .changesets/breaking_lb_next_apollo_otel_default.md (hunk 3 above)
Comment (lennyburdette) [inline .changesets/breaking_lb_next_apollo_otel_default.md:5]:
```suggestion
```

File: .changesets/breaking_lb_next_apollo_otel_default.md (hunk 2 above)
Comment (lennyburdette) [reply .changesets/breaking_lb_next_apollo_otel_default.md:3]: great idea, i don't know the answer. it's possible that by the time 2.0 is GA, the option will be removed entirely, so i'm not going to worry about it right now

**Pull request merged**